pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7594589?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7594589?buildTab=artifacts#/) on release-22.2 @ [022d19ab6c1c6f6dd477d5120b4c47a46f744238](https://github.com/cockroachdb/cockroach/commits/022d19ab6c1c6f6dd477d5120b4c47a46f744238):


```
Slow failing tests:
TestWorkload - 895.00s

Slow passing tests:
TestWorkload - 627.58s
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92043 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-schema branch-master]
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21626

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7637754?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7637754?buildTab=artifacts#/) on release-22.2 @ [2a71a5cca1f66313fd39956f1a3db070f67ade15](https://github.com/cockroachdb/cockroach/commits/2a71a5cca1f66313fd39956f1a3db070f67ade15):


```
I221122 08:16:53.589291 383243 9@util/log/event_log.go:32  [n3,replication-reporter,intExec=get-previous-replication-stats] 92088 ={"Timestamp":1669105013586324279,"EventType":"query_execute","Statement":"SELECT zone_id, subzone_id, total_ranges, unavailable_ranges, under_replicated_ranges, over_replicated_ranges FROM \"\".system.replication_stats","Tag":"SELECT","User":"root","ApplicationName":"$ internal-get-previous-replication-stats","ExecMode":"exec-internal","NumRows":5,"Age":1.625351,"FullTableScan":true}
I221122 08:16:53.600279 383334 9@util/log/event_log.go:32  [n3,replication-reporter,intExec=timestamp-upsert-replication-stats] 92089 ={"Timestamp":1669105013586324279,"EventType":"query_execute","Statement":"UPSERT INTO \"\".system.reports_meta(id, \"generated\") VALUES ($1, $2)","Tag":"INSERT","User":"root","ApplicationName":"$ internal-timestamp-upsert-replication-stats","PlaceholderValues":["3","'2022-11-22 08:16:53.586324'"],"ExecMode":"exec-internal","NumRows":1,"Age":10.588502}
I221122 08:16:53.611044 383335 9@util/log/event_log.go:32  [n3,replication-reporter,intExec=upsert-replication-stats] 92090 ={"Timestamp":1669105013586324279,"EventType":"query_execute","Statement":"UPSERT INTO \"\".system.replication_stats(report_id, zone_id, subzone_id, total_ranges, unavailable_ranges, under_replicated_ranges, over_replicated_ranges) VALUES ($1, $2, $3, $4, $5, $6, $7)","Tag":"INSERT","User":"root","ApplicationName":"$ internal-upsert-replication-stats","PlaceholderValues":["3","0","0","66","0","0","0"],"ExecMode":"exec-internal","NumRows":1,"Age":10.306731}
I221122 08:16:53.620797 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92091 ={"Timestamp":1669105013612517613,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fe26e9bf0000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":8.274279,"TxnCounter":2503}
I221122 08:16:53.628090 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92092 ={"Timestamp":1669105013621489756,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'978344db-427f-4344-b9a3-5364ec7262d6'"],"ExecMode":"exec","NumRows":1,"Age":6.58825,"TxnCounter":2505}
I221122 08:16:54.097058 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92093 ={"Timestamp":1669105014088861719,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fec2dd830000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":8.185268,"TxnCounter":2507}
I221122 08:16:54.105214 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92094 ={"Timestamp":1669105014097666185,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'071b38be-cf62-464c-9631-0a9def127170'"],"ExecMode":"exec","NumRows":1,"Age":7.535312,"TxnCounter":2509}
I221122 08:16:54.636798 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92095 ={"Timestamp":1669105014630001066,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fe26e9bf0000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":6.784306,"TxnCounter":2511}
I221122 08:16:54.647699 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92096 ={"Timestamp":1669105014637570530,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'978344db-427f-4344-b9a3-5364ec7262d6'"],"ExecMode":"exec","NumRows":1,"Age":10.114357,"TxnCounter":2513}
I221122 08:16:55.113497 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92097 ={"Timestamp":1669105015105981702,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fec2dd830000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":7.516655,"TxnCounter":2515}
I221122 08:16:55.121192 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92098 ={"Timestamp":1669105015114479097,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'071b38be-cf62-464c-9631-0a9def127170'"],"ExecMode":"exec","NumRows":1,"Age":7.039264,"TxnCounter":2517}
I221122 08:16:55.439318 383650 13@kv/kvserver/replicate_queue.go:1613  [n1,replicate,s1,r49/1:/Table/{48-50}] 92099  no suitable rebalance target for voters
I221122 08:16:55.439399 383650 13@kv/kvserver/replicate_queue.go:1633  [n1,replicate,s1,r49/1:/Table/{48-50}] 92100  no suitable rebalance target for non-voters
I221122 08:16:55.439500 383650 13@kv/kvserver/replicate_queue.go:1862  [n1,replicate,s1,r49/1:/Table/{48-50}] 92101  transferring lease to s2
I221122 08:16:55.657185 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92102 ={"Timestamp":1669105015649160069,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fe26e9bf0000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":8.021646,"TxnCounter":2519}
I221122 08:16:55.664753 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92103 ={"Timestamp":1669105015657995689,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'978344db-427f-4344-b9a3-5364ec7262d6'"],"ExecMode":"exec","NumRows":1,"Age":6.740494,"TxnCounter":2521}
I221122 08:16:55.958228 383353 9@util/log/event_log.go:32  [n1,intExec=get-jobs] 92104 ={"Timestamp":1669105012600710314,"EventType":"query_execute","Statement":"SELECT id, payload FROM \"\".system.jobs WHERE status IN ('running', 'pending', 'cancel-requested', 'pause-requested', 'reverting', 'paused') ORDER BY created","Tag":"SELECT","User":"root","ApplicationName":"$ internal-get-jobs","ExecMode":"exec-internal","NumRows":67,"Age":3357.713,"TxnCounter":1}
I221122 08:16:55.960590 383731 9@util/log/event_log.go:32  [n1,intExec=job-row-insert] 92105 ={"Timestamp":1669105015958552940,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","Tag":"INSERT","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["816036629572976641","'running'","'\\x0a415461626c652073746174697374696373207265667265736820666f722073797374656d2e7075626c69632e7472616e73616374696f6e5f737461746973746963731204726f6f748201594352454154452053544154495354494353205f5f6175746f5f5f2046524f4d205b34335d2057495448204f5054494f4e53205448524f54544c494e4720302e39204153204f462053595354454d2054494d4520272d333073279a0210f1b6c9abc49c49e68ac23e027b47f9a3a2020808161002180020007a860f0a085f5f6175746f5f5f12fa0c0a167472616e73616374696f6e5f73746174697374696373182b200128013a0a08b68ec7e2bdaef6941742330a0d616767726567617465645f747310011a0d080910001800300050a00960002000300068007000780080010088010098010042330a0e66696e6765727072696e745f696410021a0c08081000180030005011600020003000680070007800800100880100980100422d0a086170705f6e616d6510031a0c08071000180030005019600020003000680070007800800100880100980100422c0a076e6f64655f696410041a0c0801104018003000501460002000300068007000780080010088010098010042380a0c6167675f696e74657276616c10051a13080610001800300050a20960006a040800100020003000680070007800800100880100980100422e0a086d6574616461746110061a0d081210001800300050da1d60002000300068007000780080010088010098010042300a0a7374617469737469637310071a0d081210001800300050da1d60002000300068007000780080010088010098010042cf010a43637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f3810081a0c080110201800300050176000200030015a656d6f6428666e76333228637264625f696e7465726e616c2e646174756d735f746f5f627974657328616767726567617465645f74732c206170705f6e616d652c2066696e6765727072696e745f69642c206e6f64655f696429292c20383a3a3a494e543829680070007800800100880100980100480952fb020a077072696d617279100118012243637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f38220d616767726567617465645f7473220e66696e6765727072696e745f696422086170705f6e616d6522076e6f64655f69642a0c6167675f696e74657276616c2a086d657461646174612a0a7374617469737469637330083001300230033004400040004000400040004a10080010001a00200028003000380040005a007005700670077a0408002000800100880100900104980101a2017b08011243637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f381808220d616767726567617465645f747322086170705f6e616d65220e66696e6765727072696e745f696422076e6f64655f6964a80100b20100ba0100c00100c80100d00101e001005a7b0a1566696e6765727072696e745f73746174735f69647810021800220e66696e6765727072696e745f69643002380838013803380440004a10080010001a00200028003000380040005a007a0408002000800100880100900103980100a20106080012001800a80100b20100ba0100c00100c80100d00100e0010060036a210a0b0a0561646d696e102018200a0a0a04726f6f741020182012046e6f64651802800101880103980100a201ef010a9701637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f3820494e2028303a3a3a494e54382c20313a3a3a494e54382c20323a3a3a494e54382c20333a3a3a494e54382c20343a3a3a494e54382c20353a3a3a494e54382c20363a3a3a494e54382c20373a3a3a494e5438291249636865636b5f637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f3818002808300038014002b201b8010a077072696d61727910001a43637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f381a0d616767726567617465645f74731a0e66696e6765727072696e745f69641a086170705f6e616d651a076e6f64655f69641a0c6167675f696e74657276616c1a086d657461646174611a0a73746174697374696373200820012002200320042005200620072800b80101c20100e80100f2010408001200f801008002009202009a020a08b68ec7e2bdaef69417b20200b80200c0021dc80200e00200800300880303a80300b003001a080a0108100120c8011a080a0101100120c8011a040a0201081a080a0102100120c8011a050a030102081a080a0103100120c8011a060a04010203081a080a0104100120c8011a070a0501020304081a070a0105100120021a050a010620021a050a0107200222594352454154452053544154495354494353205f5f6175746f5f5f2046524f4d205b34335d2057495448204f5054494f4e53205448524f54544c494e4720302e39204153204f462053595354454d2054494d4520272d333073272a0a08aee3d7e5b6c4f69417322473797374656d2e7075626c69632e7472616e73616374696f6e5f7374617469737469637339cdccccccccccec3f4001'","'\\x1099d8c6dcacc1fb027a00'","'\\x3a69f421986345e3b86ed3a44d51909e'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":1.900927}
I221122 08:16:55.989722 383697 9@util/log/event_log.go:32  [n1,intExec=select-job] 92106 ={"Timestamp":1669105015971352244,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id, COALESCE(last_run, created), COALESCE(num_runs, 0) FROM \"\".system.jobs WHERE id = $1","Tag":"SELECT","User":"root","ApplicationName":"$ internal-select-job","PlaceholderValues":["816036629572976641"],"ExecMode":"exec-internal","NumRows":1,"Age":18.277285}
I221122 08:16:55.992757 383748 9@util/log/event_log.go:32  [n1,intExec=job-update] 92107 ={"Timestamp":1669105015971352244,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","Tag":"UPDATE","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["816036629572976641","'\\x1098bcc7dcacc1fb02a801b9a6e2879f9cb5cd6c7a00'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.594525}
I221122 08:16:55.999893 383696 jobs/registry.go:1211  [n1] 92108  AUTO CREATE STATS job 816036629572976641: stepping through state running
I221122 08:16:56.024884 382519 9@util/log/event_log.go:32  [n3,intExec=expire-sessions] 92109 ={"Timestamp":1669105005986611179,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET claim_session_id = NULL WHERE claim_session_id IN (SELECT claim_session_id WHERE ((claim_session_id != $1) AND (status IN ('running', 'pending', 'cancel-requested', 'pause-requested', 'reverting'))) AND (NOT crdb_internal.sql_liveness_is_alive(claim_session_id)) LIMIT $2)","Tag":"UPDATE","User":"root","ApplicationName":"$ internal-expire-sessions","PlaceholderValues":["'\\xe91ea1e3946e45cd94b7b32b3079cde8'","1000"],"ExecMode":"exec-internal","Age":10038.19}
I221122 08:16:56.025971 383592 9@util/log/event_log.go:32  [n2,intExec=expire-sessions] 92110 ={"Timestamp":1669105015279166848,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET claim_session_id = NULL WHERE claim_session_id IN (SELECT claim_session_id WHERE ((claim_session_id != $1) AND (status IN ('running', 'pending', 'cancel-requested', 'pause-requested', 'reverting'))) AND (NOT crdb_internal.sql_liveness_is_alive(claim_session_id)) LIMIT $2)","Tag":"UPDATE","User":"root","ApplicationName":"$ internal-expire-sessions","PlaceholderValues":["'\\xa754d14d68f84b73ac042c218537a0d9'","1000"],"ExecMode":"exec-internal","Age":746.7272}
I221122 08:16:56.027788 383752 9@util/log/event_log.go:32  [n3,intExec=cancel/pause-requested] 92111 ={"Timestamp":1669105016025305313,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = CASE WHEN status = 'pause-requested' THEN 'paused' WHEN status = 'cancel-requested' THEN 'reverting' ELSE status END, num_runs = 0, last_run = NULL, claim_session_id = NULL, claim_instance_id = NULL WHERE (status IN ('pause-requested', 'cancel-requested')) AND ((claim_session_id = $1) AND (claim_instance_id = $2)) RETURNING id, status","Tag":"UPDATE","User":"node","ApplicationName":"$ internal-cancel/pause-requested","PlaceholderValues":["'\\xe91ea1e3946e45cd94b7b32b3079cde8'","3"],"ExecMode":"exec-internal","Age":2.423882}
I221122 08:16:56.028087 383658 9@util/log/event_log.go:32  [n2,intExec=cancel/pause-requested] 92112 ={"Timestamp":1669105016026211016,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = CASE WHEN status = 'pause-requested' THEN 'paused' WHEN status = 'cancel-requested' THEN 'reverting' ELSE status END, num_runs = 0, last_run = NULL, claim_session_id = NULL, claim_instance_id = NULL WHERE (status IN ('pause-requested', 'cancel-requested')) AND ((claim_session_id = $1) AND (claim_instance_id = $2)) RETURNING id, status","Tag":"UPDATE","User":"node","ApplicationName":"$ internal-cancel/pause-requested","PlaceholderValues":["'\\xa754d14d68f84b73ac042c218537a0d9'","2"],"ExecMode":"exec-internal","Age":1.817997}
I221122 08:16:56.035108 383751 9@util/log/event_log.go:32  [n1,intExec=select-job] 92113 ={"Timestamp":1669105015999949209,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id, COALESCE(last_run, created), COALESCE(num_runs, 0) FROM \"\".system.jobs WHERE id = $1","Tag":"SELECT","User":"root","ApplicationName":"$ internal-select-job","PlaceholderValues":["816036629572976641"],"ExecMode":"exec-internal","NumRows":1,"Age":35.109337}
I221122 08:16:56.038645 383519 9@util/log/event_log.go:32  [n1,intExec=job-update] 92114 ={"Timestamp":1669105015999949209,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, last_run = $4, num_runs = $5 WHERE id = $1","Tag":"UPDATE","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["816036629572976641","'running'","'\\x0a415461626c652073746174697374696373207265667265736820666f722073797374656d2e7075626c69632e7472616e73616374696f6e5f737461746973746963731204726f6f7418a6b0cbdcacc1fb028201594352454154452053544154495354494353205f5f6175746f5f5f2046524f4d205b34335d2057495448204f5054494f4e53205448524f54544c494e4720302e39204153204f462053595354454d2054494d4520272d333073279a0210f1b6c9abc49c49e68ac23e027b47f9a3a2020808161002180020007a860f0a085f5f6175746f5f5f12fa0c0a167472616e73616374696f6e5f73746174697374696373182b200128013a0a08b68ec7e2bdaef6941742330a0d616767726567617465645f747310011a0d080910001800300050a00960002000300068007000780080010088010098010042330a0e66696e6765727072696e745f696410021a0c08081000180030005011600020003000680070007800800100880100980100422d0a086170705f6e616d6510031a0c08071000180030005019600020003000680070007800800100880100980100422c0a076e6f64655f696410041a0c0801104018003000501460002000300068007000780080010088010098010042380a0c6167675f696e74657276616c10051a13080610001800300050a20960006a040800100020003000680070007800800100880100980100422e0a086d6574616461746110061a0d081210001800300050da1d60002000300068007000780080010088010098010042300a0a7374617469737469637310071a0d081210001800300050da1d60002000300068007000780080010088010098010042cf010a43637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f3810081a0c080110201800300050176000200030015a656d6f6428666e76333228637264625f696e7465726e616c2e646174756d735f746f5f627974657328616767726567617465645f74732c206170705f6e616d652c2066696e6765727072696e745f69642c206e6f64655f696429292c20383a3a3a494e543829680070007800800100880100980100480952fb020a077072696d617279100118012243637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f38220d616767726567617465645f7473220e66696e6765727072696e745f696422086170705f6e616d6522076e6f64655f69642a0c6167675f696e74657276616c2a086d657461646174612a0a7374617469737469637330083001300230033004400040004000400040004a10080010001a00200028003000380040005a007005700670077a0408002000800100880100900104980101a2017b08011243637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f381808220d616767726567617465645f747322086170705f6e616d65220e66696e6765727072696e745f696422076e6f64655f6964a80100b20100ba0100c00100c80100d00101e001005a7b0a1566696e6765727072696e745f73746174735f69647810021800220e66696e6765727072696e745f69643002380838013803380440004a10080010001a00200028003000380040005a007a0408002000800100880100900103980100a20106080012001800a80100b20100ba0100c00100c80100d00100e0010060036a210a0b0a0561646d696e102018200a0a0a04726f6f741020182012046e6f64651802800101880103980100a201ef010a9701637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f3820494e2028303a3a3a494e54382c20313a3a3a494e54382c20323a3a3a494e54382c20333a3a3a494e54382c20343a3a3a494e54382c20353a3a3a494e54382c20363a3a3a494e54382c20373a3a3a494e5438291249636865636b5f637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f3818002808300038014002b201b8010a077072696d61727910001a43637264625f696e7465726e616c5f616767726567617465645f74735f6170705f6e616d655f66696e6765727072696e745f69645f6e6f64655f69645f73686172645f381a0d616767726567617465645f74731a0e66696e6765727072696e745f69641a086170705f6e616d651a076e6f64655f69641a0c6167675f696e74657276616c1a086d657461646174611a0a73746174697374696373200820012002200320042005200620072800b80101c20100e80100f2010408001200f801008002009202009a020a08b68ec7e2bdaef69417b20200b80200c0021dc80200e00200800300880303a80300b003001a080a0108100120c8011a080a0101100120c8011a040a0201081a080a0102100120c8011a050a030102081a080a0103100120c8011a060a04010203081a080a0104100120c8011a070a0501020304081a070a0105100120021a050a010620021a050a0107200222594352454154452053544154495354494353205f5f6175746f5f5f2046524f4d205b34335d2057495448204f5054494f4e53205448524f54544c494e4720302e39204153204f462053595354454d2054494d4520272d333073272a0a08aee3d7e5b6c4f69417322473797374656d2e7075626c69632e7472616e73616374696f6e5f7374617469737469637339cdccccccccccec3f4001'","'2022-11-22 08:16:56.035367'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":3.139562}
I221122 08:16:56.139271 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92115 ={"Timestamp":1669105016123126903,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fec2dd830000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":16.139463,"TxnCounter":2523}
I221122 08:16:56.147211 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92116 ={"Timestamp":1669105016139935242,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'071b38be-cf62-464c-9631-0a9def127170'"],"ExecMode":"exec","NumRows":1,"Age":7.256584,"TxnCounter":2525}
I221122 08:16:56.673971 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92117 ={"Timestamp":1669105016666297486,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fe26e9bf0000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":7.666765,"TxnCounter":2527}
I221122 08:16:56.684113 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92118 ={"Timestamp":1669105016674670573,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'978344db-427f-4344-b9a3-5364ec7262d6'"],"ExecMode":"exec","NumRows":1,"Age":9.429442,"TxnCounter":2529}
I221122 08:16:57.045614 383877 9@util/log/event_log.go:32  [n1,intExec=expire-sessions] 92119 ={"Timestamp":1669105017042916820,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET claim_session_id = NULL WHERE claim_session_id IN (SELECT claim_session_id WHERE ((claim_session_id != $1) AND (status IN ('running', 'pending', 'cancel-requested', 'pause-requested', 'reverting'))) AND (NOT crdb_internal.sql_liveness_is_alive(claim_session_id)) LIMIT $2)","Tag":"UPDATE","User":"root","ApplicationName":"$ internal-expire-sessions","PlaceholderValues":["'\\x3a69f421986345e3b86ed3a44d51909e'","1000"],"ExecMode":"exec-internal","Age":2.626316}
I221122 08:16:57.048303 383890 9@util/log/event_log.go:32  [n1,intExec=cancel/pause-requested] 92120 ={"Timestamp":1669105017045908928,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = CASE WHEN status = 'pause-requested' THEN 'paused' WHEN status = 'cancel-requested' THEN 'reverting' ELSE status END, num_runs = 0, last_run = NULL, claim_session_id = NULL, claim_instance_id = NULL WHERE (status IN ('pause-requested', 'cancel-requested')) AND ((claim_session_id = $1) AND (claim_instance_id = $2)) RETURNING id, status","Tag":"UPDATE","User":"node","ApplicationName":"$ internal-cancel/pause-requested","PlaceholderValues":["'\\x3a69f421986345e3b86ed3a44d51909e'","1"],"ExecMode":"exec-internal","Age":2.330264}
I221122 08:16:57.155410 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92121 ={"Timestamp":1669105017148642424,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fec2dd830000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":6.767268,"TxnCounter":2531}
I221122 08:16:57.165376 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92122 ={"Timestamp":1669105017156099631,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'071b38be-cf62-464c-9631-0a9def127170'"],"ExecMode":"exec","NumRows":1,"Age":9.249695,"TxnCounter":2533}
I221122 08:16:57.692428 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92123 ={"Timestamp":1669105017685748973,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fe26e9bf0000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":6.677469,"TxnCounter":2535}
I221122 08:16:57.699611 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92124 ={"Timestamp":1669105017693129159,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'978344db-427f-4344-b9a3-5364ec7262d6'"],"ExecMode":"exec","NumRows":1,"Age":6.47443,"TxnCounter":2537}
I221122 08:16:58.174136 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92125 ={"Timestamp":1669105018167129907,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fec2dd830000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":6.992974,"TxnCounter":2539}
I221122 08:16:58.196829 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92126 ={"Timestamp":1669105018174902673,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'071b38be-cf62-464c-9631-0a9def127170'"],"ExecMode":"exec","NumRows":1,"Age":21.91095,"TxnCounter":2541}
I221122 08:16:58.679847 501 2@util/log/event_log.go:32  [n1] 92127 ={"Timestamp":1669105018679843258,"EventType":"runtime_stats","MemRSSBytes":4765777920,"GoroutineCount":1313,"MemStackSysBytes":21430272,"GoAllocBytes":2177884640,"GoTotalBytes":4455919224,"GoStatsStaleness":0.001309189,"HeapFragmentBytes":370966048,"HeapReservedBytes":1662042112,"HeapReleasedBytes":528211968,"CGoAllocBytes":309103304,"CGoTotalBytes":371286016,"CGoCallRate":2443.17,"CPUUserPercent":134.51486,"CPUSysPercent":59.40656,"GCPausePercent":0.03060154,"GCRunCount":221,"NetHostRecvBytes":61341392,"NetHostSendBytes":61341392}
I221122 08:16:58.680064 501 2@server/status/runtime_log.go:47  [n1] 92128  runtime stats: 4.4 GiB RSS, 1313 goroutines (stacks: 20 MiB), 2.0 GiB/4.1 GiB Go alloc/total(stale) (heap fragmentation: 354 MiB, heap reserved: 1.5 GiB, heap released: 504 MiB), 295 MiB/354 MiB CGO alloc/total (2443.2 CGO/sec), 134.5/59.4 %(u/s)time, 0.0 %gc (221x), 58 MiB/58 MiB (r/w)net
I221122 08:16:58.707716 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92129 ={"Timestamp":1669105018700449182,"EventType":"query_execute","Statement":"SELECT active_queries, kv_txn FROM crdb_internal.cluster_sessions WHERE session_id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'1729d974fe26e9bf0000000000000001'"],"ExecMode":"exec","NumRows":1,"Age":7.254081,"TxnCounter":2543}
I221122 08:16:58.715311 8437 9@util/log/event_log.go:32  [n1,client=127.0.0.1:47038,user=testuser] 92130 ={"Timestamp":1669105018708330926,"EventType":"query_execute","Statement":"SELECT sum(num_retries) + sum(num_auto_retries) FROM crdb_internal.cluster_transactions WHERE id = $1","Tag":"SELECT","User":"testuser","ApplicationName":"schemachange","PlaceholderValues":["'978344db-427f-4344-b9a3-5364ec7262d6'"],"ExecMode":"exec","NumRows":1,"Age":6.96494,"TxnCounter":2545}
panic: connection has timed out &{0xc000fd1650 0xc00574f800 1729d974fe26e9bf0000000000000001 0xc05c5aaf00 {CREATE TABLE schema1295.table1300 AS SELECT public.table1036.col738_746, public.table1036.col738_751, public.table1036.col738_741, public.table1036.col738_740, public.table1036.col738_749, public.table1036.col738_752, public.table1036.col738_745, public.table1036.col781_782 FROM public.table1036 LIMIT 300000 true} {978344db-427f-4344-b9a3-5364ec7262d6 true} 0}

goroutine 210159 [running]:
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWatchDog).watchLoop(0xc04bce4060, {0x6288410, 0xc00728a200})
	github.com/cockroachdb/cockroach/pkg/workload/schemachange/watch_dog.go:116 +0x26d
created by github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWatchDog).Start
	github.com/cockroachdb/cockroach/pkg/workload/schemachange/watch_dog.go:134 +0x139
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92043 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-schema branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go
+++ b/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go
@@ -89,7 +89,7 @@ func TestWorkload(t *testing.T) {
 	ql, err := wl.Ops(ctx, []string{pgURL.String()}, reg)
 	require.NoError(t, err)
 
-	const N = 800
+	const N = 100
 	workerFn := func(ctx context.Context, fn func(ctx context.Context) error) func() error {
 		return func() error {
 			for i := 0; i < N; i++ {

